### ISSUE TITLE
[🐛] Cannot create property 'path' on string

### ISSUE THREAD

**Issue by hyperlink** [snyk/snyk#1377]
- `node -v`: v10.20.1
- `npm -v`: 6.14.4
- `snyk -v`: 1.387.0
- OS: (e.g. OSX, Linux, Windows, ...)
- Command run: `npx snyk monitor --org=realtime`

### Expected behaviour

monitor runs without issue.

### Actual behaviour

```
14:20:18 Monitoring /var/build/workspace/build-<project>/apps/server...
14:20:18 
14:20:18 Cannot create property 'path' on string '����n� �߅�b���[��M���h�*SK��1����nK��w�wΏ�s�9��QG��6 ��{#x@0�OB\r6�`�bCBB�JaZ�Q�+V�:�cu!a��l����z%���	h�ɕ0��~����!����޵��yS�!Ŧ���l��m���Xg�;�����2BAG|�V�����&���S�ϋ�H�n����%;`7�a�r��KWO���9K6�H�����H���|��ke��,���o,.�r��I$g����*��-wp.�S��0��Gz�v����41�ՙ�m;�M��	��Y���o�������`;JY�!Lp�Yq����Ru>���$�UZF0�h�S��(�j�(
14:20:18 p��;$��'
14:20:18 
```

### Steps to reproduce
Please share _minimal_ steps needed to reproduce your issue. Ideally
a paired down manifest / project to showcase the problem that can also
be used for testsing.

### Debug log
```
  snyk monitor { _: [ [Circular] ],
  debug: true,
  org: 'realtime',
  callGraphBuilderTimeout: '300' } +0ms
  snyk Processing /Users/xiaoxin/dev/rt/apps/server... +0ms
  snyk getDepsFromPlugin ... +2ms
  snyk-test Analyzing npm dependencies for /Users/xiaoxin/dev/rt/apps/server/package-lock.json +0ms
  snyk Processing realtime... +68ms
  snyk not using proxy +0ms
  snyk Potential policy locations found: [ '/Users/xiaoxin/dev/rt/apps/server' ] +0ms
  snyk sending request to: https://snyk.io/api/v1/monitor/npm/graph +1s
  snyk request body size: 131260 +0ms
  snyk gzipped request body size: 18885 +0ms
  snyk not using proxy +1ms
Error:
Monitoring /Users/xiaoxin/dev/rt/apps/server...

Cannot create property 'path' on string '���n� �߅k���n�f7mf�^���2�4�K㻏:분:{|��8�C��;�u�	�j�2���4A��=㌂OJ\��x�pB����Q����Յ���
x�ۋ
z!��E�	i�%���^9���[_c���7eҧ�t��mwS�-�2B���y��IlI�������!����&�ˤc�ϳ�@�n����%�:`7�a�|h�s�L�9[�
                                                                                              �H�����H�`Q�誮� ��r>1���2(GY�0��lV1�W������a�R�*����-�D�&�4:�i��nwS!=�^�^��T��}�|�8.#�J��8�Iy��pok�@���8��5�A��p�c���p��W:�D���'
    at monitor (/Users/xiaoxin/.npm/_npx/54810/lib/node_modules/snyk/src/cli/commands/monitor/index.ts:298:9)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  snyk analytics { args:
   [ { debug: true, org: 'realtime', callGraphBuilderTimeout: '300' } ],
  command: 'bad-command',
  org: 'realtime',
  metadata:
   { pluginOptions:
      { debug: true, org: 'realtime', callGraphBuilderTimeout: '300' },
     local: true,
     'generating-node-dependency-tree': { lockFile: true, targetFile: 'package-lock.json' },
     pluginName: 'snyk-nodejs-lockfile-parser',
     packageManager: [ 'npm', 'npm' ],
     isDocker: false,
     experimentalMonitorDepGraphFromDepTree: true,
     policies: 1,
     policyLocations: [ '/Users/xiaoxin/dev/rt/apps/server' ],
     targetBranch: 'nda-workaround',
     payloadSize: 131260,
     gzippedPayloadSize: 18885,
     'error-message':
      '\nMonitoring /Users/xiaoxin/dev/rt/apps/server...\n\nCannot create property \'path\' on string \'\u001f�\b\u0000\u0000\u0000\u0000\u0000\u0000\u0003���n� \u0014�߅k�\u0014��n�f7mf�^���2�4�K\u001a㻏:분:{\u0007|��8�C�\u0013�;�\u0019u�\t�j�2���\u00004A��=㌂OJ\\��\u0015\u000ex�pB�\u0000���Q\u0018\u0013����Յ���\nx�ۋ\nj\rz!��E�\ti�\u0004%�\u0000��^\u00169���[_c��\u0006�7eҧ�t�\u0015�mwS�-�2B�\u0016��y�\u001d�\u0006Il\u0004I\u0019���\u0006����!\u001e����\u001f&�ˤc�ϳ�@�n����%�:`7�a�|h�s�L�9[�\u000b�H����\u001d�H�`Q�誮\u0019� �\u001f�r>1\u001f���2(GY�0��lV1�W�\u0012��\u000e���a�R\u001c�*��\u001e��-�D�&�4:�i��nwS!=�^�^��\u0007T��}�|�8.#\u0006�J�\b��\u00048�\u0011Iy\u001c��pok�@��K\u0013\b�8�\u0018�5�A��p�c���p��W\u0001:\u0001�D���\u0005\u0000\u0000\'',
     error:
      'Error: \nMonitoring /Users/xiaoxin/dev/rt/apps/server...\n\nCannot create property \'path\' on string \'\u001f�\b\u0000\u0000\u0000\u0000\u0000\u0000\u0003���n� \u0014�߅k�\u0014��n�f7mf�^���2�4�K\u001a㻏:분:{\u0007|��8�C�\u0013�;�\u0019u�\t�j�2���\u00004A��=㌂OJ\\��\u0015\u000ex�pB�\u0000���Q\u0018\u0013����Յ���\nx�ۋ\nj\rz!��E�\ti�\u0004%�\u0000��^\u00169���[_c��\u0006�7eҧ�t�\u0015�mwS�-�2B�\u0016��y�\u001d�\u0006Il\u0004I\u0019���\u0006����!\u001e����\u001f&�ˤc�ϳ�@�n����%�:`7�a�|h�s�L�9[�\u000b�H����\u001d�H�`Q�誮\u0019� �\u001f�r>1\u001f���2(GY�0��lV1�W�\u0012��\u000e���a�R\u001c�*��\u001e��-�D�&�4:�i��nwS!=�^�^��\u0007T��}�|�8.#\u0006�J�\b��\u00048�\u0011Iy\u001c��pok�@��K\u0013\b�8�\u0018�5�A��p�c���p��W\u0001:\u0001�D���\u0005\u0000\u0000\'\n    at monitor (/Users/xiaoxin/.npm/_npx/54810/lib/node_modules/snyk/src/cli/commands/monitor/index.ts:298:9)\n    at process._tickCallback (internal/process/next_tick.js:68:7)',
     'error-code': undefined,
     command: 'monitor' },
  version: '1.387.0',
  os: 'macOS Catalina',
  nodeVersion: 'v10.20.1',
  standalone: false,
  integrationName: '',
  integrationVersion: '',
  id: 'fb5b58b5e20f6fe2725134a58c258a300549778c',
  ci: false,
  durationMs: 6419 } +0ms
  snyk sending request to: https://snyk.io/api/v1/analytics/cli +5s
  snyk request body size: 2530 +0ms
  snyk gzipped request body size: 1111 +0ms
  snyk not using proxy +0ms

  snyk Exit code: 2 +0ms
```

**Comment by postmechanical** [comment]
Running into this exact same issue as well.

**Comment by JackuB** [comment]
You are right, there seems to be a regression. We are looking into the cause

**Comment by JackuB** [comment]
It seems to be caused by an issue in an HTTP client [needle](https://www.npmjs.com/package/needle)

<img width="1240" alt="screenshot_2020-09-01_at_23 27 59" src="https://user-images.githubusercontent.com/1788727/91908230-84cfe000-ecab-11ea-9b22-472fe93fb45a.png">

We are looking into our option

**Comment by JackuB** [comment]
We've just released 1.387.1 https://github.com/snyk/snyk/releases/tag/v1.387.1 - with older version of `needle`. Updating to that will fix the `snyk monitor` issue.

Also, once the needle@2.5.2 is published, this issue should also go away even for older versions.

**Comment by hyperlink** [comment]
The latest version works. thanks for the quick turnaround!
